### PR TITLE
(PC-7593) App native - Inscription - Arrêt du parcours ID Check à l'étape de profil

### PR DESCRIPTION
**Pull request by kopax-polyconseil**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-7593

## Checklist

I have:

- [ ] Made sure the title of my PR follows the convention `[PC-XXX] <summary>`.
- [ ] Made sure my feature is working on the relevant real / virtual devices.
- [ ] Written **unit tests** for my feature.
- [ ] Added new reusable components to the AppComponents page and communicate to the team on slack
- [ ] Added a **screenshot** for UI tickets.
- [ ] Attached a **ticket number** for any added TODO/FIXME \
       (for tech tasks, give the best context about the TODO resolution: what/who/when).

## Screenshots

| \*iOS - [Phone name] | \*Android - [Phone name] | Tablet - [Phone name] | Little - [Phone name] |
| -------------------- | :----------------------: | --------------------: | --------------------: |
|                      |                          |                       |                       |

## Deploy hard

If native code (ios/android) was modified, **after** the PR is merged, on the master branch, upgrade the **app version** (+1 patch):

- if you want an hard deployment of the testing environment, use `yarn trigger:testing:deploy`

### CODE REVIEW

File: src/features/auth/hooks/useKeyboardAdjustFixIdCheck.android.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import { useKeyboardAdjust } from 'ui/components/keyboard/useKeyboardAdjust'
+
+/*
+ * On Android, webview crash when opening/closing keyboard with adjustMode=resize. (https://passculture.atlassian.net/browse/PC-7593)
+ * It crash when user animate the keyboard in IDcheck when it's open from SignupConfirmationEmailSent
+ * Bellow fix will disable keyboard animation and inject css over webview js api to move up on the screen all the buttons which would normally be hidden by it.
```
Comment (BenStirrup) [inline src/features/auth/hooks/useKeyboardAdjustFixIdCheck.android.ts:6]:
```suggestion
 * On Android, webview crashes when opening/closing keyboard with `adjustMode=resize` : (https://passculture.atlassian.net/browse/PC-7593)
 * When does it happen in user path : when using the signup-confirmation deeplink right after seeing the  SignupConfirmationEmailSent screen.
 * The fix below will disable keyboard animation (only on Android) and inject css over the webview's JavaScript API to move up on the screen all the buttons which would normally be hidden by the keyboard.
```

File: src/features/auth/hooks/useKeyboardAdjustFixIdCheck.android.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  '.form.form { justify-content: initial !important; }',
+  '.form.form > button.primary { margin-top: 2em !important; }',
+];
+
+// Load the rules and execute after the DOM loads
+window.onload = function() { 
+  addCss(rules.join('\\n'));
+};
+})();`
+}
```
Comment (BenStirrup) [inline src/features/auth/hooks/useKeyboardAdjustFixIdCheck.android.ts:35]: Du javascript dans une string. Hum... 😄

Comment (kopax-polyconseil) [reply src/features/auth/hooks/useKeyboardAdjustFixIdCheck.android.ts:35]: Que proposes tu ? J'avais eu la même remarque sur le `.html`, mais je ne crois pas qu'on ai des loaders comme dans webpack.

Comment (BenStirrup) [reply src/features/auth/hooks/useKeyboardAdjustFixIdCheck.android.ts:35]: J'ai rien à proposer, donc ca me va

**Review by BenStirrup — APPROVED** [review]

**Pull request merged**